lighthouse-core/lib/page-functions.js

```diff
@@ -225,6 +232,44 @@ function getNodeSelector(node) {
   return parts.join(' > ');
 }
 
+/**
+ * This function checks if an element or an ancestor of an element is `position:fixed`.
+ * In addition we ensure that the element is capable of behaving as a `position:fixed`
+ * element, checking that it lives within a scrollable ancestor.
+ * @param {HTMLElement} element
+ * @return {boolean}
+ */
+/* istanbul ignore next */
+function isPositionFixed(element) {
+  /**
+   * @param {HTMLElement} element
+   * @param {string} attr
+   * @return {string}
+   */
+  function getStyleAttrValue(element, attr) {
+    // Check style before computedStyle as computedStyle is expensive.
+    return element.style[attr] || window.getComputedStyle(element)[attr];
+  }
+
+  // Position fixed/sticky has no effect in case when document does not scroll.
```

Comment:
FWIW, I didn't mean to discourage the entire pattern since there are legitimate cases where it will have an effect and the document isn't scrollable. 

My primary goal was to see what we cared about for the `isPositionFixed` in `adspeed` since I expected those cases to be rarer in iframes and making sure we documented this function accordingly with what to look out for :)

Reply:
Gotcha, after your comment though I realized that this really is the only case we need to worry about.

Our motivation for including this is specifically for our ad-top-of-viewport audit. We're telling publishers to not put an ad at the top of the viewport because it can be scrolled past before loading, however, in the case of a fixed ad this advice is irrelevant.